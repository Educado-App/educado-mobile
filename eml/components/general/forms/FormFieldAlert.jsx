import React from 'react';
import { View } from 'react-native';
import Text from '../Text';
import PropTypes from 'prop-types';

/**
 * Component for showing an alert below a form field
 * @param {Object} props should contain the following properties:
 * - label: String
 * - success: Boolean
 * @returns {React.Element} JSX element for showing alerts
 */
export default function FormFieldAlert(props) {
	return (
		<View className="flex-row items-center">
			{props.success ?
				<Text className="text-xs text-success mx-2">{props.label}</Text>
				: <Text className="text-xs text-error mx-2">{props.label}</Text>}
		</View >
	);
}

FormFieldAlert.propTypes = {
	label: PropTypes.string,
	success: PropTypes.bool,
};