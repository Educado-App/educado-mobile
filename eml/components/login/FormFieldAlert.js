import { React } from 'react';
import { Text, View } from 'react-native';

export default function FormFieldAlert(props) {
  return (
    <View className="flex-row items-center">
      <Text className="text-xs text-error mx-2 font-montserrat">
        {/* Passwords must match */}
        {props.label}
      </Text>
    </View>
  )
}