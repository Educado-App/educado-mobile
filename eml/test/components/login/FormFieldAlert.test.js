import renderer from "react-test-renderer";
import React from "react";
import FormFieldAlert from "../../../components/general/forms/FormFieldAlert";

test('Ensure that the alert is rendered correctly', async () => {
  let formAlert;
  await renderer.act(() => {
    formAlert = renderer.create(<FormFieldAlert />);
  });
  let tree = formAlert.toJSON();
  expect(tree).toMatchSnapshot();
});