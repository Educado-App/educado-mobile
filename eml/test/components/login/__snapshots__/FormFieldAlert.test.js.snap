// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensure that the alert is rendered correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Montserrat-Regular",
        },
        {
          "color": "#383838",
        },
        {
          "fontSize": 16,
        },
        [
          {
            "fontSize": 12,
            "lineHeight": 16,
          },
          {
            "color": "#FF4949",
          },
          {
            "marginLeft": 8,
            "marginRight": 8,
          },
        ],
      ]
    }
  />
</View>
`;
